File: Python/isPower.py
class Solution:
    # @param A : integer
    # @return a boolean
    def isPower(self, N):
        if N == 0:
            return False
        if N == 1:
            return True
        for p in xrange(2,33):
            for A in xrange(2, int(N**(1.0 / p)) + 2):
                if A**p == N:
                    return True
        return False

File: C++/isPower.cpp
class Solution {
public:
  bool isPower(int x) {
	if (x <= 1) return true;
	for (int base = 2; base < x && base < INT_MAX / base; base++) {
	  int temp = base;
	  while (temp <= x && temp < INT_MAX / base) {
		temp *= base;
		if (temp == x) return true;
	  }
	}
	return false;
  }
};
